lib/src/distance.dart

```diff
@@ -0,0 +1,365 @@
+          ? origin?.map((o) => Uri.encodeComponent(o))?.join('|')
+          : origin,
+      "destinations": destination != null && destination is String
+          ? destination?.map((d) => Uri.encodeComponent(d))?.join('|')
+          : destination,
+      "mode": travelModeToString(travelMode),
+      "language": languageCode,
+      "region": region,
+      "avoid": routeTypeToString(routeType),
+      "units": unitToString(unit),
+      "arrival_time": arrivalTime is DateTime
+          ? arrivalTime.millisecondsSinceEpoch ~/ 1000
+          : arrivalTime,
+      "departure_time": departureTime is DateTime
+          ? departureTime.millisecondsSinceEpoch ~/ 1000
+          : departureTime,
+      "traffic_model": trafficModelToString(trafficModel),
+      "transit_mode":
+          transitMode?.map((t) => transitModeToString(t))?.join("|"),
+      "transit_routing_preference":
+          transitRoutingPreferencesToString(transitRoutingPreference)
+    };
+
+    if (apiKey != null) {
+      params.putIfAbsent("key", () => apiKey);
+    }
+
+    return '$url?${buildQuery(params)}';
+  }
+
+  DistanceResponse _decode(Response res) =>
+      new DistanceResponse.fromJson(json.decode(res.body));
+}
+
+class DistanceResponse extends GoogleResponseStatus {
+  final List<String> originAddress;
+  final List<String> destinationAddress;
+  final List<Row> results;
+
+  DistanceResponse(
+    String status,
+    String errorMsg,
+    this.originAddress,
+    this.destinationAddress,
+    this.results,
+  ) : super(
+          status,
+          errorMsg,
+        );
+
+  factory DistanceResponse.fromJson(Map json) => new DistanceResponse(
+      json["status"],
+      json["error_message"],
+      (json["origin_addresses"] as List)?.cast<String>(),
+      (json["destination_addresses"] as List)?.cast<String>(),
+      json["rows"]
+          ?.map((row) {
+            return new Row.fromJson(row);
+          })
+          ?.toList()
+          ?.cast<Row>());
+}
+
+class Row {
+  final List<Element> elements;
+
+  Row(this.elements);
+
+  factory Row.fromJson(Map json) => json != null
+      ? new Row(json["elements"]
+          ?.map((element) {
+            return new Element.fromJson(element);
+          })
+          ?.toList()
+          ?.cast<Element>())
+      : null;
+}
+
+class Element {
+  final Value distance;
+  final Value duration;
+  final String elementStatus;
+
+  Element(this.distance, this.duration, this.elementStatus);
+
+  factory Element.fromJson(Map json) => json != null
+      ? new Element(
+          new Value.fromJson(json["distance"]),
+          new Value.fromJson(json["duration"]),
+          json["status"],
+        )
+      : null;
+}
+
+class Value {
+  final num value;
+  final String text;
+
+  Value(this.value, this.text);
+
+  factory Value.fromJson(Map json) =>
+      json != null ? new Value(json["value"], json["text"]) : null;
+}
+
+enum TravelMode { driving, walking, bicycling, transit }
```

Comment:
Following enums already exist in the directions api, can you move them in `lib/src/core.dart` to avoid duplicate ?

Reply:
```suggestion
enum TravelMode { driving, walking, bicycling, transit }
```
I have also move these enums to core.dart because direction and distance both are using these enum and some helper methods 
`enum TransitMode { bus, subway, train, tram, rail }`
`enum TrafficModel { bestGuess, pessimistic, optimistic }`
`enum RouteType { tolls, highways, ferries, indoor }`
`enum TravelMode { driving, walking, bicycling, transit }`

some methods such as this one
`TravelMode stringToTravelMode(String mode) {....}`